pipeline.py

```diff
@@ -88,12 +103,27 @@ def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+
+    if not path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    if not any(Path(path_to_validate).iterdir()):
+        raise EmptyDirectoryError
+
+    file_formats = [".json", ".txt", ".pdf"]
+    files = path_to_validate.iterdir()
```

Comment:
there is no guaranteed order in what iterdir returns to you. therefore it is naive to expect that it will always return a list from 1 to N